### PR TITLE
awsls: init at 0.11.0

### PR DESCRIPTION
**Pull request by markus1189**
###### Description of changes

New package: awsls to list aws resources, homepage: https://github.com/jckuester/awsls

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/awsls/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner = "jckuester";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-iy9tohmVUtNXYVfe6pZ+pbbLlcK6Fu1GgzTWMD+3xP0=";
+  };
+
+  vendorSha256 = "sha256-ZyMO+KCqoePF6MqHFt8X4tZR4nBhuSPgJDrX+emM6jc=";
+
+  ldflags = [ "-s" "-w" "-X github.com/jckuester/awsls/internal.version=${version}" "-X github.com/jckuester/awsls/internal.commit=${src.rev}" "-X github.com/jckuester/awsls/internal.date=unknown" ];
```
Comment (sikmir) [inline pkgs/tools/admin/awsls/default.nix:16]:
```suggestion
  ldflags = let t = "github.com/jckuester/awsls/internal"; in [ "-s" "-w" "-X ${t}.version=${version}" "-X ${t}.commit=${src.rev}" "-X ${t}.date=unknown" ];
```

Comment (markus1189) [reply pkgs/tools/admin/awsls/default.nix:16]: Hm yes  we could DRY  that up a bit,  although I'm okay with being explicit  here.  Why  did you choose `t` as the variable name?  Maybe  `prefix` would  be  better?

Comment (SuperSandro2000) [reply pkgs/tools/admin/awsls/default.nix:16]: IDK, t is short and used in other places

Comment (markus1189) [reply pkgs/tools/admin/awsls/default.nix]: Done

**Pull request merged**